star-dash/star-dash/GameBridge/SyncModule/ObjectModule.swift

```diff
@@ -0,0 +1,31 @@
+import CoreGraphics
+import SDPhysicsEngine
+
+class ObjectModule: SyncModule {
+    let entityManager: EntitySyncInterface
+
+    init(entityManager: EntitySyncInterface) {
+        self.entityManager = entityManager
+    }
+
+    func sync(entity: Entity, from object: SDObject) {
+        guard let positionComponent = entityManager.component(ofType: PositionComponent.self, of: entity.id) else {
```

Comment:
Should the syncing be done through system manager? I did create a sync function in the position and physics systems.

Reply:
I believe this approach makes for better design since our game engine has one less responsibility and reduced coupling to the physics engine.